### DISCUSSION TITLE
Is it possible to download streams with Streamlink while using this client?

### DISCUSSION THREAD

**Discussion by UltraX7** [streamlink/streamlink-twitch-gui#970] [Q&A] [answered]
I was wondering if it's possible to include arguments for Streamlink to write the stream you're watching through this client to a file, or even just use the client as a lightweight means to know when they go live and write them to files for later viewing, even without being able to watch them simultaneously. 

As far as I can tell, it only seems to be possible to include arguments for external players that the client launches, and not Streamlink itself, but if anyone knows better and can help me that'd be awesome. Thanks :)

**Comment by bastimeyer** [comment, marked as answer]
> As far as I can tell, it only seems to be possible to include arguments for external players that the client launches, and not Streamlink itself

Enable advanced settings in the main settings menu and then you're able to set custom streamlink arguments in the streaming settings menu. There are no additional checks for which arguments are supported, etc, so if you set incompatible ones, then you won't be able to launch streams.
https://streamlink.github.io/cli.html

And in regards to recording, this would only work with the `--record` argument, not with `--output`, because the Twitch GUI expects a player to be launched.

**Reply by UltraX7** [reply]
Ah! Don't know how I missed that tbh lol, many thanks :P
